geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanner.scala

```diff
@@ -210,6 +213,33 @@ case class QueryPlanner(schema: String,
     }
   }
 
+  def adaptIteratorForMapAggregationQuery(accumuloIterator: CloseableIterator[Entry[Key, Value]],
+                                          query: Query,
+                                          returnSFT: SimpleFeatureType,
+                                          decoder: SimpleFeatureDecoder): CloseableIterator[SimpleFeature] = {
+    val aggregateKeyName = query.getHints.get(MAP_AGGREGATION_KEY).asInstanceOf[String]
+
+    val maps = accumuloIterator.map { kv =>
+      decoder
+        .decode(kv.getValue.get)
+        .getAttribute(aggregateKeyName)
+        .asInstanceOf[JMap[AnyRef, Int]]
```

Comment:
question: could we support the other numeric types instead of just int? I'm assuming that Int is the thing for this use case but if it were some other measurement stored in a Map it may be a double or something.